src/chain/chain.ts

```diff
@@ -58,14 +70,26 @@ export class BeaconChain extends EventEmitter implements IBeaconChain {
     }
   }
 
-  public async stop(): Promise<void> {}
+  public async stop(): Promise<void> {
+  }
 
   public async initializeChain(
     genesisTime: number64,
     genesisDeposits: Deposit[],
     genesisEth1Data: Eth1Data
   ): Promise<void> {
     this.logger.info('Initializing beacon chain.');
+    const merkleTree = ProgressiveMerkleTree.empty(DEPOSIT_CONTRACT_TREE_DEPTH);
```

Comment:
I think this should be done in `eth1`.
If pre-genesis, have a merkle tree that gets deposits added when they're processed.

Reply:
Not really sure if that would optimize much, as this is only adding items to array and calculating hashes once. 
In eth1 we would have to introduce db service just for that.